client/signup/config/steps-pure.js

```diff
@@ -361,6 +361,12 @@ export function generateSteps( {
 			stepName: 'reader-landing',
 			providesDependencies: [],
 		},
+
+		/* Improved Onboarding */
+		'site-type': {
+			stepName: 'site-type',
+			providesDependencies: [ 'siteType' ],
```

Comment:
I think we should be providing `themeSlugWithRepo` (and skipping the logic in the About step if it exists) in this step, to prepare for the point where the About step won't be present anymore.

Reply:
Regarding tracks event - we probably need just the two you mentioned:
1. User selects a site (also if they can't decide and change their selection)
2. User submits moves to the next step

Reply:
> The value of themeSlugWithRepo when when submit the signup step is 'pub/radcliffe-2' when there's a landing page vertical. Do we bring all the logic over to the Site Type component?

@ramonjd: ultimately, if a user comes here from a landing page, we should be auto-setting the segment (and thus theme), and skipping this step. Verticals won't technically determine the theme, segmenting will, we just currently only have vertical landing pages trying to do this.

I guess it brings up a greater issue of how we're determining a valid landing page. I don't think we should maintain a whitelist of VLPs, since that list will grow substantially. We could just validate the `websites` vertical landing page directory (`https://wordpress.com/websites/{vertical-slug}/`) and then maintain a shorter list of other possible LPs (i.e. wordpress.com/business/). We could also just use a query string on the CTAs for landing pages that set the segment and/or vertical (i.e. wordpress.com/start/?segment=business&vertical=restaurant). What do you think?

> Also, should we provide designType as well?

This feels like a legacy setting from the old Triforce Signup step, where a user picked their "design" based on a few icons. We should audit what this actually does in Signup (if anything) and the backend, and possibly remove it entirely. I'll take a closer look this weekend, but I think we can just skip it on this step.

> is there any benefit in recording a tracks event when the user selects or submits a site type

We already have an event for `calypso_signup_step_complete` that should provide the type (via dependency submitted). I'll double-check. If we do, we can probably skip adding any other event.

Reply:
> I guess it brings up a greater issue of how we're determining a valid landing page
Could you help me understand why we need to determine this. Presently, we have a logic that if a user comes via a VLP, then we set the theme to `pub/radcliffe2` but I'm not sure I understand why we do this. Irrespective of where the user came from(VLP or just directly via wordpress.com), shouldn't the theme be selected only based on the segment? 

Or is there a plan to select a theme based on vertical+segment values? So in the future, is the idea that the default theme selected for restuarant(LP) + blogger(segment) could be different from that for Gym(LP) + blogger(segment)? If so, then a query string format seems a good suggestions.

Reply:
> We should audit what this actually does in Signup (if anything) and the backend, and possibly remove it entirely. I'll take a closer look this weekend, but I think we can just skip it on this step.

From a quick look, it looks like the `designType` is used to determine `store` flows. There could be more uses - let me know if you think we can clean this up if it's a legacy code.

Reply:
> Could you help me understand why we need to determine this. Presently, we have a logic that if a user comes via a VLP, then we set the theme to pub/radcliffe2 but I'm not sure I understand why we do this. Irrespective of where the user came from(VLP or just directly via wordpress.com), shouldn't the theme be selected based on the segment only?

@niranjan-uma-shankar the way it currently is, we're assuming a "business" type of site, based on the fact that vertical landing pages are marketed to businesses. This was before the new segment step, so the logic is murky. As you note, with the new flow, theme will be selected based on segment only. We'll just pre-select the "business" segment for users that come from landing pages marketed to businesses. I still think a query string makes sense here, since we don't want to maintain a whitelist of landing pages, and it doesn't present a security or spam risk to allow this to be passed as a ref. We'd just compare passed `?segment` values against a whitelist of the 5 segments.

> From a quick look, it looks like the designType is used to determine store flows. There could be more - let me know if you think we can clean this up if it's a legacy code.

Thanks for looking into this for me! I think we can leave the `designType` logic out of this step. It needs to remain in the About step (since the current Main flow will still route users into that flow), but in the new `onboarding` flow, we'll handle store flows in a more declarative manner (by maybe eventually adding a sixth segment of "Store" to this step).

Let me know if you have any other questions. If this changes the tasking on the card, just update it with new tasks.